apps/omg/lib/omg/state.ex

```diff
@@ -132,7 +122,15 @@ defmodule OMG.State do
   Checks (stateful validity) and executes a spend transaction. Assuming stateless validity!
   """
   def handle_call({:exec, tx, fees}, _from, state) do
-    case Core.exec(state, tx, fees) do
+    db_utxos =
+      tx
+      |> Transaction.get_inputs()
+      |> init_utxos_from_db(state)
+
+    state
+    |> Core.with_utxos(db_utxos)
```

Comment:
this seems to be a bit confusing step. why is this step separate from processing a transaction?

Reply:
Well extending the state of the utxos from DB is in several places, not just for tx processing.